pkg/openSearch/osquery/compareHandler.go

```diff
@@ -0,0 +1,208 @@
+func HandleCompareOperatorIsLessThanOrEqualTo(fieldName string, fieldValue any) (esquery.Mappable, error) {
+	if values, ok := fieldValue.([]any); ok {
+		return esquery.Bool(). // chain by OR
+					Should(
+				lo.Map(values, func(value any, _ int) esquery.Mappable {
+					return esquery.Range(fieldName).
+						Lte(value)
+				})...,
+			).
+			MinimumShouldMatch(1), nil
+	} else {
+		return esquery.Range(fieldName).
+			Lte(fieldValue), nil
+	}
+}
+
+// HandleCompareOperatorIsGreaterThanOrEqualTo handles is greater than or equal to
+func HandleCompareOperatorIsGreaterThanOrEqualTo(fieldName string, fieldValue any) (esquery.Mappable, error) {
+	if values, ok := fieldValue.([]any); ok {
+		return esquery.Bool(). // chain by OR
+					Should(
+				lo.Map(values, func(value any, _ int) esquery.Mappable {
+					return esquery.Range(fieldName).
+						Gte(value)
+				})...,
+			).
+			MinimumShouldMatch(1), nil
+	} else {
+		return esquery.Range(fieldName).
+			Gte(fieldValue), nil
+	}
+}
+
+// HandleCompareOperatorIsGreaterThan handles is greater than
+func HandleCompareOperatorIsGreaterThan(fieldName string, fieldValue any) (esquery.Mappable, error) {
+	if values, ok := fieldValue.([]any); ok {
+		return esquery.Bool(). // chain by OR
+					Should(
+				lo.Map(values, func(value any, _ int) esquery.Mappable {
+					return esquery.Range(fieldName).
+						Gt(value)
+				})...,
+			).
+			MinimumShouldMatch(1), nil
+	} else {
+		return esquery.Range(fieldName).
+			Gt(fieldValue), nil
+	}
+}
+
+// HandleCompareOperatorIsLessThan handles is less than
+func HandleCompareOperatorIsLessThan(fieldName string, fieldValue any) (esquery.Mappable, error) {
+	if values, ok := fieldValue.([]any); ok {
+		return esquery.Bool(). // chain by OR
+					Should(
+				lo.Map(values, func(value any, _ int) esquery.Mappable {
+					return esquery.Range(fieldName).
+						Lt(value)
+				})...,
+			).
+			MinimumShouldMatch(1), nil
+	} else {
+		return esquery.Range(fieldName).
+			Lt(fieldValue), nil
+	}
+}
+
+func HandleCompareOperatorExists(fieldName string, _ any) (esquery.Mappable, error) {
+	return esquery.Exists(fieldName), nil
+}
+
+// HandleCompareOperatorBetweenDates constructs an OpenSearch range query for a given date field.
+// It accepts a field name and a field value, which must be either:
+// - A slice of two time.Time values ([]time.Time), representing the start and end of the range, or
+// - A slice of two RFC3339Nano-formatted strings ([]string), which are parsed into time.Time,  representing the start and end of the range.
+//
+// The generated range query is inclusive of both the lower and upper bounds.
+// If a document’s timestamp is exactly equal to the start or end date, it will still match the query.
+//
+// If the slice length is not exactly 2, or if the string values cannot be parsed into valid dates,
+// the function logs an error and returns an empty query (MatchNone).
```

Comment:
The doc can be adjusted

```suggestion
// HandleCompareOperatorBetweenDates constructs an OpenSearch range query for a given date field.
// It accepts a field name and a field value, which must be a slice of exactly 2 elements, representing the start and end of range. Accepted slice types:
// - []time.Time,
// - []string of two RFC3339Nano-formatted strings,
// - []any, containing any combination of time.Time and RFC3339Nano-formatted string.
//
// The generated range query is inclusive of both the lower and upper bounds.
// If a document’s timestamp is exactly equal to the start or end date, it will still match the query.
```